packages/node-utils/src/http.ts

```diff
@@ -231,8 +304,19 @@ export class HttpServer<T extends EventMap> extends TypedEmitter<T & BaseEvents>
             ...route.handler,
         ];
 
-        const run = ([handler, ...rest]: Handler[]) => {
-            handler?.(request as Request, response, () => run(rest), { logger: this.logger });
+        const run = ([handler, ...rest]: Handler<RequestWithParams>[]) => {
+            handler?.(
+                requestWithParams,
+                response,
+                (nextRequest, nextResponse) => {
+                    requestWithParams = nextRequest;
+                    response = nextResponse;
+                    run(rest);
+                },
+                {
+                    logger: this.logger,
+                },
+            );
         };
 
         run(handlers);
```

Comment:
Does it work the same? Tests are passing. No reassigning to neither `requestWithParams` nor `response` though.
```suggestion
        const run =
            ([handler, ...rest]: Handler<RequestWithParams>[]) =>
            (req: RequestWithParams, res: http.ServerResponse) =>
                handler?.(req, res, run(rest), { logger: this.logger });

        run(handlers)(requestWithParams, response);
```

Reply:
[9e5ad43](https://github.com/trezor/trezor-suite/pull/9985/commits/9e5ad43d856d6d2dbed31bb60e1177bb7210f7ac)